src/render.js

```diff
@@ -1,48 +1,40 @@
-import entries from 'object.entries'
-
-const {hasOwnProperty} = Object.prototype
-const tags = {}
-let prevStyles = {}
+const tags = new Map()
+let prevStyles = new Map()
 
 export default typeof window === 'undefined' ? renderOnServer : renderOnClient
 
 function renderOnServer() {}
 
-function renderOnClient(components) {
-  const styles = {}
-  for (const c of components) {
-    styles[c.props.styleId] = c
```

Comment:
The use case is for global styles like css resets which you want to initialise at the top of your component tree and render before anything else.